File: subprojects/java-compatibility/src/test/java/kotools/types/number/ZeroTest.java
package kotools.types.number;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ZeroTest {
    @Test
    public void toString_should_be_compatible_with_Java() {
        final String actual = Zero.INSTANCE.toString();
        final String expected = "0";
        final String message = "String representation of Zero should be '0'.";
        Assertions.assertEquals(expected, actual, message);
    }
}

File: v5/src/commonTest/kotlin/org/kotools/types/number/ZeroTest.kt
package org.kotools.types.number

import org.kotools.types.ExperimentalApi
import kotlin.test.Test
import kotlin.test.assertEquals

class ZeroTest {
    @OptIn(ExperimentalApi::class)
    @Test
    fun toString_should_pass() {
        val actual: String = Zero.toString()
        val expected = "0"
        val message = "'Zero.toString()' should return '$expected'."
        assertEquals(expected, actual, message)
    }
}
